File: Joueur.ts/src/games/chess/ai.ts
// This is where you build your AI for the Chess game.

import { BaseAI } from "../../joueur/base-ai";
import { Game } from "./game";
import { Player } from "./player";

// <<-- Creer-Merge: imports -->>
// any additional imports you want can be required here safely between creer runs
// <<-- /Creer-Merge: imports -->>

/**
 * This is the class to play the Chess game.
 * This is where you should build your AI.
 */
export class AI extends BaseAI {
    /**
     * The reference to the Game instance this AI is playing.
     */
    public readonly game!: Game;

    /**
     * The reference to the Player this AI controls in the Game.
     */
    public readonly player!: Player;
    /**
     * This is the name you send to the server so your AI
     * will control the player named this string.
     *
     * @returns A string for the name of your player.
     */
    public getName(): string {
        // <<-- Creer-Merge: getName -->>
        return "Chess JavaScript Player";
        // <<-- /Creer-Merge: getName -->>
    }

    /**
     * This is called once the game starts and your AI knows its playerID and game.
     * You can initialize your AI here.
     */
    public async start(): Promise<void> {
        // <<-- Creer-Merge: start -->>
        // pass
        // <<-- /Creer-Merge: start -->>
    }

    /**
     * This is called every time the game's state updates, so if you are tracking anything you can update it here.
     */
    public async gameUpdated(): Promise<void> {
        // <<-- Creer-Merge: gameUpdated -->>
        // pass
        // <<-- /Creer-Merge: gameUpdated -->>
    }

    /**
     * This is called when the game ends, you can clean up your data and dump files here if need be.
     *
     * @param won True means you won, false means you lost.
     * @param reason The human readable string explaining why you won or lost.
     */
    public async ended(won: boolean, reason: string): Promise<void> {
        // <<-- Creer-Merge: ended -->>
        // pass
        // <<-- /Creer-Merge: ended -->>
    }
    /**
     * This is called every time it is this AI.player's turn to make a move.
     * @returns A string in Universal Chess Interface (UCI) or Standard
     * Algebraic Notation (SAN) formatting for the move you want to make. If the
     * move is invalid or not properly formatted you will lose the game.
     */
    public async makeMove(): Promise<string> {
        // <<-- Creer-Merge: makeMove -->>
        // Put your game logic here for makeMove
        return "";
        // <<-- /Creer-Merge: makeMove -->>
    }

    // <<-- Creer-Merge: functions -->>
    // any additional functions you want to add for your AI
    // <<-- /Creer-Merge: functions -->>
}

File: Joueur.go/games/jungle_chess/ai.go
package jungleChess

import "joueur/base"

// PlayerName should return the string name of your Player in games it plays.
func PlayerName() string {
	// <<-- Creer-Merge: getName -->>
	return "JungleChess Go Player"
	// <<-- /Creer-Merge: getName -->>
}

// AI is your personal AI implimentation.
type AI struct {
	base.AIImpl
	// <<-- Creer-Merge: fields -->>
	// You can add new fields here
	// <<-- /Creer-Merge: fields -->>
}

// Game returns the instance of the Game this AI is currently playing.
func (ai *AI) Game() Game {
	return ai.AIImpl.Game().(Game)
}

// Player returns the instance of the Player this AI is represented by in the
// game this AI is playing.
func (ai *AI) Player() Player {
	return ai.AIImpl.Player().(Player)
}

// Start is called once the game starts and your AI has a Player and Game.
// You can initialize your AI struct here.
func (ai *AI) Start() {
	// <<-- Creer-Merge: start -->>
	// pass
	// <<-- /Creer-Merge: start -->>
}

// GameUpdated is called every time the game's state updates,
// so if you are tracking anything you can update it here.
func (ai *AI) GameUpdated() {
	// <<-- Creer-Merge: game-updated -->>
	// pass
	// <<-- /Creer-Merge: game-updated -->>
}

// Ended is called when the game ends, you can clean up your data and dump
// files here if need be.
func (ai *AI) Ended(won bool, reason string) {
	// <<-- Creer-Merge: ended -->>
	// pass
	// <<-- /Creer-Merge: ended -->>
}

// -- JungleChess specific AI actions -- \\

// MakeMove this is called every time it is this AI.player's turn to make a
// move.
func (ai *AI) MakeMove() string {
	// <<-- Creer-Merge: makeMove -->>
	// Put your game logic here for makeMove
	return ""
	// <<-- /Creer-Merge: makeMove -->>
}
